db/model/Variable.ts

```diff
@@ -141,65 +147,56 @@ export async function getVariableData(
             additionalInfo: partialSource.additionalInfo || "",
         },
     }
-    const variableData: OwidVariableMixedData = {
-        years: [],
-        entities: [],
-        values: [],
-    }
 
-    const entityMap = new Map<number, OwidVariableDimensionValueFull>()
-    const yearMap = new Map<number, OwidVariableDimensionValuePartial>()
+    const entities = await db.queryMysql(
+        `
+    SELECT
+        id,
+        name,
+        code
+    FROM entities WHERE id in (?)
+    `,
+        [_.uniq(variableData.entities)]
+    )
+
+    const years = _.uniq(variableData.years).map((year) => ({ id: year }))
 
+    return {
+        ...variableMetadata,
+        dimensions: {
+            years: { values: years },
+            entities: { values: entities },
+        },
+    }
+}
+
+function detectValuesType(
+    values: (string | number)[]
+): OwidVariableTypeOptions {
```

Comment:
Would be great if we can add tests!

Reply:
ChatGPT can for sure ;) (By the way all this code should soon go away and as far as I know, we don't use the type anyway)